project/Build.scala

```diff
@@ -489,7 +490,7 @@ object Build {
       // get libraries onboard
       libraryDependencies ++= Seq(
         "org.scala-lang.modules" % "scala-asm" % "7.3.1-scala-1", // used by the backend
-        Dependencies.`compiler-interface`,
+        Dependencies.oldCompilerInterface, // we stick to the old version to avoid deprecation warnings
```

Comment:
What's the plan for removing these warnings? Should we open an issue to keep track of it?

Reply:
The plan is to upgrade the version of Zinc in Mill. Then use the new Zinc API in the compiler. I have a branch with those changes in my fork that has passed the CI, except for Mill community projects.

I'll open an issue as soon as this is merged.